File: src/main/java/cz/ememsoft/minibank/dto/CustomerDto.java
package cz.ememsoft.minibank.dto;

import cz.ememsoft.minibank.entity.Customer;
import jakarta.validation.constraints.NotNull;

import java.io.Serializable;
import java.util.List;

/**
 * DTO for {@link Customer}
 */
public record CustomerDto(Long id, String first_name, String middle_name, String last_name,
                          @NotNull List<AccountDto> account) implements Serializable {
}

File: src/main/kotlin/cz/ememsoft/minibank/api/dto/CustomerDto.kt
package cz.ememsoft.minibank.api.dto

data class CustomerDto(
    val firstName: String,
    val lastName: String,
    val email: String,
    val phone: String,
    val address: String,
) {
    override fun toString(): String {
        return "CustomerDto(firstName='$firstName', lastName='$lastName', email='$email', phone='$phone', address='$address')"
    }
}
